src/EVM/Expr.hs

```diff
@@ -685,9 +685,14 @@ simplify e = if (mapExpr go e == e)
       | a == b = Lit 1
       | otherwise = Lit 0
     go (Eq (Lit 0) (Sub a b)) = Eq a b
-    go o@(Eq a b)
-      | a == b = Lit 1
-      | otherwise = o
+    go (Eq a b@(Lit _)) = Eq b a
+    go (Eq a b) =
+      case (aSimp, bSimp) of
```

Comment:
Is this needed? Don't we already call `simplify` everywhere until fixpoint as part of `mapExpr`?

Reply:
Oh of course, what was I thinking. Sorry.